### ISSUE TITLE
parser: redeclare variable with function declaration inside the function should not throw error

### ISSUE THREAD

**Issue by Dunqing** [oxc-project/oxc#10159]
```js
// Valid
function a() { var b; function b() { } }`

// Invalid
function a() { class b { }; function b() { } }
```
Reporting by https://github.com/oxc-project/monitor-oxc/actions/runs/14189330473/job/39750446597
